adapter/config/memory/config.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package memory provides an in-memory volatile config store implementation
+package memory
+
+import (
+	"errors"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+
+	"istio.io/pilot/model"
+)
+
+// Make creates an in-memory config store from a config descriptor
+func Make(descriptor model.ConfigDescriptor) model.ConfigStore {
+	out := store{
+		descriptor: descriptor,
+		data:       make(map[string]map[string]proto.Message),
+		revs:       make(map[string]map[string]string),
+	}
+	for _, typ := range descriptor.Types() {
+		out.data[typ] = make(map[string]proto.Message)
+		out.revs[typ] = make(map[string]string)
+	}
+	return &out
+}
+
+type store struct {
+	descriptor model.ConfigDescriptor
+	data       map[string]map[string]proto.Message
```

Comment:
I don't see concurrent use in our code. By default, nothing is thread-safe, so it's ok to skip mutex for now.